### DISCUSSION TITLE
Recommendations for going about 3D pathfinding utilizing JitterPhysics2

### DISCUSSION THREAD

**Discussion by captainkidd5** [notgiven688/jitterphysics2#100] [How do I ... ?]
Hello! I was wondering if there are any recommended libraries which might play nicely with JP2, or if there might be a simple way to leverage the octree in JP2 for something like A*

**Comment by captainkidd5** [comment]
On another note, is there anything which might be a close fit for 

https://docs.unity3d.com/ScriptReference/Physics.CheckSphere.html ?

I'm porting over https://github.com/TheCyaniteProject/PathFinder3D/blob/main/Assets/Resources/Scripts/AstarGrid.cs to Monogame and the only line which doesn't have a close match with jitter would be 

`  node.walkable = !Physics.CheckSphere(node.position, colliderRadius, obsticleMask);
`

**Comment by notgiven688** [comment]
Hello, based on my understanding, you're interested in implementing a pathfinding algorithm using the octree implementation found in JitterDemo. It's important to note that Octree.cs is not a part of the physics engine; rather, it's a class I created to showcase the engine's ability to handle custom collision detection. Feel free to incorporate Octree.cs into your projects. It's worth mentioning that Jitter employs a different internal acceleration structure known as a dynamic AABB-tree.

In reference to the Unity documentation on [Physics.CheckSphere](https://docs.unity3d.com/ScriptReference/Physics.CheckSphere.html), this method returns true if there is any collidable object within the radius of a sphere. Adding a similar functionality to Octree.cs should be relatively straightforward. You can implement this by creating a method analogous to [InternalQuery(Stack<uint> triangles, in JBBox box, uint nodeIndex)](https://github.com/notgiven688/jitterphysics2/blob/da8903d11ff0c419a336d0f328837772bd7dd4e3/src/JitterDemo/Demos/CustomCollision/Octree.cs#L98). The `InternalQuery` method collects all triangles intersecting the specified `box`.

**Reply by captainkidd5** [reply]
Ok that method was what I was looking for, thank you!
